src/lambda_handler.py

```diff
@@ -1,89 +1,231 @@
+
+        # Fill document fields with CVE data
+        print(".", end="")
+
         try:
-            variable_error_msg = "numerator: %s, denominator: %s"
-            result["result"] = example_div(int(numerator), int(denominator))
+            cve_doc = CVEDoc(
+                id=cve_id,
+                cvss_score=float(cvss_base_score),
+                cvss_version=cvss_version_temp,
+                severity=None,
+            )
         except ValueError:
-            error_msg = "The provided values must be integers."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
-        except ZeroDivisionError:
-            error_msg = "The denominator cannot be zero."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
+            logger.error("CVE object: %s", cve)
+            raise ValueError("JSON does not look like valid NVD CVE data.")
+
+        await cve_doc.save()
+
+        return cve_doc.id
+
+
+async def process_cve_json(json_stream) -> None:
+    """Process the provided CVEs JSONs and update the database with their contents."""
+    data = json.load(json_stream)
+
+    if data.get("CVE_data_type") != "CVE":
+        raise ValueError("JSON does not look like valid NVD CVE data.")
+
+    tasks = [asyncio.create_task(process_nvd(cve)) for cve in data.get("CVE_Items", [])]
+    await asyncio.gather(*tasks)
+
+
+async def unzip_and_process(content):
+    """Unzip content and process it with process_cve_json()."""
+    with gzip.GzipFile(fileobj=BytesIO(content)) as f:
+        await process_cve_json(f)
+
+
+async def fetch_url_content(session, url):
+    """Fetch the content of the provided URL."""
+    async with session.get(url) as response:
+        return await response.read()
+
+
+async def process_urls(cve_urls, db) -> None:
+    """Initialize Beanie ODM and begin processing CVE URLs."""
+    await init_beanie(database=motor_client[db], document_models=[CVEDoc])
+
+    # Create a new session for each URL to avoid throttling
+    async with aiohttp.ClientSession() as session:
+        tasks = []
+
+        for cve_url in cve_urls:
+            attempts = 0
+            retries = 10
 
-    return result
+            while True:
+                try:
+                    print("Before socket")
+                    content = await fetch_url_content(session, cve_url)
+                    tasks.append(asyncio.create_task(unzip_and_process(content)))
+                    break
+                except aiohttp.ClientError as err:
+                    logging.debug(
+                        "Encountered a(n) %s exception while attempting to fetch URL '%s'",
+                        type(err).__name__,
+                        cve_url,
+                    )
+                    attempts += 1
+                    await asyncio.sleep(5)
```

Comment:
The `5` here should be turned into a variable (e.g. `CVE_URL_RETRY_WAIT_SEC`) and moved up top (by `DEFAULT_NVD_URL` and friends) for easier updates.

Reply:
@dav3r I addressed this here: https://github.com/cisagov/cyhy-nvdsync-lambda/pull/2/commits/d184fbd02d1b43d67c1c9cfa71d4f8b3fb08319e